app-datepicker.js

```diff
@@ -0,0 +1,1756 @@
+      '_computeSeparateFormat(format)',
+      '_computeShowLongDate(showLongDate, locale)',
+      '_updateToReflectExternalChange(inputDate)',
+      '_updateThemeColor(theme)',
+      '_updateDatepickerView(view)',
+      '_computeDaysOfWeek(firstDayOfWeek, locale)',
+      '_computeActiveMonthYear(_activeYear, _activeMonth, locale)',
+      '_computeShortSelectedDate(_selectedYear, _selectedMonth, _selectedDate, locale)',
+      '_computeShowSelectedYear(_selectedYear, locale)',
+      '_computeShiftedDisableDays(firstDayOfWeek, disableDays.*)',
+    ]
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    if (!this.noAnimation) {
+      this.set('animationConfig', {
+        'incrementEntry': [
+          {
+            name: 'slide-from-right-animation',
+            node: this.$.daysOfWeekAnimation
+          },
+          {
+            name: 'slide-from-right-animation',
+            node: this.$.daysOfMonthAnimation
+          },
+          {
+            name: 'datepicker-slide-from-right-animation',
+            node: this.$.navMonthYearAnimation
+          }
+        ],
+        'decrementEntry': [
+          {
+            name: 'slide-from-left-animation',
+            node: this.$.daysOfWeekAnimation
+          },
+          {
+            name: 'slide-from-left-animation',
+            node: this.$.daysOfMonthAnimation
+          },
+          {
+            name: 'datepicker-slide-from-left-animation',
+            node: this.$.navMonthYearAnimation
+          }
+        ]
+      });
+      this.set('pageEntryAnimation', 'fade-in-animation');
+      this.set('pageExitAnimation', 'fade-out-animation');
+    }
+
+    // NOTE: workaround to check for effectiveChildren[0] as paper-button outside of
+    // this element will be queried as well even though outside of content tag.
+    // Setup distributed children.
+    var effectiveChildren = this.getEffectiveChildren();
+    if (effectiveChildren && effectiveChildren.length > 0) {
+      for (var i = 0; i < effectiveChildren.length; i++) {
+        // addEventListener to paper-button with dialog-confirm.
+        if (effectiveChildren[i].hasAttribute('dialog-confirm')) {
+          // attach event handler which first binded to this scope.
+          effectiveChildren[i].addEventListener('tap', this._updateBindDate.bind(this));
+          effectiveChildren[i].addEventListener('transitionend', this._updateBindDate.bind(this));
+        }
+        // Polymer's polyfill for custom properties does not apply to distributed children.
+        // https://www.polymer-project.org/1.0/docs/devguide/styling.html
+        this._updateDistributedButtonInkColorCustomProp(effectiveChildren[i], '#737373');
+      }
+      // update to a new height for datepicker if paper-buttons present.
+      this.$.dp.classList.add('with-buttons');
+    } else {
+      this.$.dp.classList.remove('with-buttons');
+    }
+
+    this.fire('app-datepicker-attached');
```

Comment:
This inherits the old `attached` callback from WC v0. This should be rename to `app-datepicker-connected`.